packages/graphql/lib/src/core/query_options.dart

```diff
@@ -194,12 +207,25 @@ typedef dynamic UpdateQuery(
 /// options for fetchmore operations
 class FetchMoreOptions {
   FetchMoreOptions({
-    this.document,
+    @Deprecated('Use documentNode instead') String document,
+    this.documentNode,
     this.variables = const <String, dynamic>{},
     @required this.updateQuery,
-  }) : assert(updateQuery != null);
+  })  : assert((document != null && documentNode == null) ||
+            (document == null && documentNode != null)),
+        assert(updateQuery != null);
+
+  DocumentNode documentNode;
```

Comment:
Why not use getters and setters?

Reply:
Added the missing initialisation to the constructor